spec/API_specification/dataframe_api/__init__.py

```diff
@@ -234,3 +235,21 @@ def is_dtype(dtype: DType, kind: str | tuple[str, ...]) -> bool:
     -------
     bool
     """
+
+def date(year: int, month: int, day: int) -> Scalar:
+    """
+    Create date object which can be used for filtering.
+
+    The full 32-bit signed integer range of dates should be supported.
```

Comment:
Given construction happens in pieces I think we should clarify this:
```suggestion
    The full 32-bit signed integer range of days since epoch should be supported (between -5877641-06-23 and 5881580-07-11 inclusive).
```